Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -419,37 +453,90 @@ public class GetGcsObjectCmdlet : GcsCmdlet
+        /// returned.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string Delimiter { get; set; }
+
         protected override void ProcessRecord()
         {
             base.ProcessRecord();
 
-            ObjectsResource.GetRequest getReq = Service.Objects.Get(Bucket, ObjectName);
-            getReq.Projection = ObjectsResource.GetRequest.ProjectionEnum.Full;
-            try
+            if (ObjectName != null)
             {
-                Object gcsObject = getReq.Execute();
-                WriteObject(gcsObject);
+                if (Delimiter != null || Prefix != null)
+                {
+                    WriteWarning("-Delimiter and -Prefix parameters will be ignored since -ObjectName is given.");
+                }
+                ObjectsResource.GetRequest getReq = Service.Objects.Get(Bucket, ObjectName);
+                getReq.Projection = ObjectsResource.GetRequest.ProjectionEnum.Full;
+                try
+                {
+                    Object gcsObject = getReq.Execute();
+                    WriteObject(gcsObject);
+                }
+                catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+                {
+                    ErrorRecord errorRecord = new ErrorRecord(
+                        new ItemNotFoundException($"Storage object '{ObjectName}' does not exist."),
+                        "ObjectNotFound",
+                        ErrorCategory.ObjectNotFound,
+                        ObjectName);
+                    ThrowTerminatingError(errorRecord);
+                }
             }
-            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+            else
             {
-                ErrorRecord errorRecord = new ErrorRecord(
-                    new ItemNotFoundException($"Storage object '{ObjectName}' does not exist."),
-                    "ObjectNotFound",
-                    ErrorCategory.ObjectNotFound,
-                    ObjectName);
-                ThrowTerminatingError(errorRecord);
+                ObjectsResource.ListRequest listReq = Service.Objects.List(Bucket);
+                listReq.Projection = ObjectsResource.ListRequest.ProjectionEnum.Full;
+                listReq.Delimiter = Delimiter;
+                listReq.Prefix = Prefix;
+                listReq.MaxResults = 100;
+
+                // When used with WriteObject, expand the IEnumerable rather than
+                // returning the IEnumerable itself. IEnumerable<T> vs. IEnumerable<IEnumerable<T>>.
+                const bool enumerateCollection = true;
+
+                // First page.
+                Objects gcsObjects = listReq.Execute();
+                WriteObject(gcsObjects.Items, enumerateCollection);
+
+                // Keep paging through results as necessary.
+                while (!Stopping && gcsObjects.NextPageToken != null)
```

Comment:
It's a property of the PowerShell cmdlet class. It is true if the cmdlet execution stopped (like if user press Ctrl + C).